### PR TITLE
Load Jquery from CDN only when using a remote site

### PR DESCRIPTION
**Pull request by eksperimental**
A javascript function "should_request_from_CDN" has been created to determine
whether to load from the CDN or from a local copy.
The rules are. if the page is served via http or https, and the host is
not a private network (127.x.x.x, or 192.168.x.x., or others), and
the hostname is contains a "."  (to avoid loading from localhost and others
user may have configured), only then, it will load from the CDN.

Range of private networks I am using: https://en.wikipedia.org/wiki/Private_network#Private_IPv4_address_spaces

I have used parts of this code, to determine if it's a private network
https://gist.github.com/dperini/729294

The other change is that now it requests from "//" and not from "https://"
*****

A second commit removes something that was remaning from I don't know where
https://github.com/elixir-lang/ex_doc/commit/1bede82875d7a17bfb3642a4e44835f5ab831488

### CODE REVIEW

**Comment by josevalim** [comment]
I will leave this up to you and @milmazz!

**Comment by Dignifiedquire** [comment]
I'm suggesting to load Jquery always from the local bundle when we are using a build system as this pulls all JavaScript into one file.

**Comment by milmazz** [comment]
IMHO, this seems a little hack (that try to cover an exceptional case) and I don't really see the *benefit from the user point of view*. Maybe I'm missing something here.

Right now, if I'm offline, I cannot reach `ajax.googleappis.com`, in that case we load jQuery from the local file and with that we can provide all the features that the `app.js` file provides. Also, we need to consider that @Dignifiedquire is working on a build system than concatenates & minifies all the JS files. So, in that case jQuery will be always loaded from the local file.

**Comment by eksperimental** [comment]
@milmazz what it does is when you are in an private network, or on your own computer (either http://localhost/, 127.0.0.1, or any non-qualified host name), or loading it from **file://**, it saves the the unnecessary request to load it from the CDN.
(and when using ladoing it from file://, the request doesn't get cached, at least I don't see it cached).

**Comment by josevalim** [comment]
Given that we will be minifying everything on our own already, maybe we don't really need this as we will already be serving a single JS file?

**Comment by eksperimental** [comment]
@josevalim It all depends on how long it will take for 0.9 to be released.
Personally I think we should not avoid changes because it will be obsolete in the new version.
I think at least 0.8.1 should be released with the fixes we are going to be added in the meantime 0.9 get ready.

**Comment by josevalim** [comment]
@eksperimental I agree with that but given that we are not so sure about the provided function too, I think we can wait. :) Unless this was a clear 100% win, then I would have no objections. :)

**Comment by eksperimental** [comment]
@josevalim We can leave it to catch only the scheme and if there is a "." in the hostname.

    ( protocol == 'http:' || protocol == 'https:' ) && host.indexOf('.') !== -1 )

or even this, if you want to still load from CDN when developing in localhost/

    ( protocol == 'http:' || protocol == 'https:' )

My point is, calling https when viewing the docs from file:// is a clear loss to me.
But we can wait :-)
